common/transport/transport-api/src/main/java/org/thingsboard/server/common/transport/adaptor/JsonConverter.java

```diff
@@ -414,22 +414,20 @@ public static JsonObject toJson(ProvisionDeviceResponseMsg payload, int requestI
 
     private static JsonObject toJson(ProvisionDeviceResponseMsg payload, boolean toGateway, int requestId) {
         JsonObject result = new JsonObject();
-        if (payload.getProvisionResponseStatus() == TransportProtos.ProvisionResponseStatus.NOT_FOUND) {
+        if (payload.getStatus() == TransportProtos.ProvisionResponseStatus.NOT_FOUND) {
             result.addProperty("errorMsg", "Provision data was not found!");
-            result.addProperty("provisionDeviceStatus", ProvisionResponseStatus.NOT_FOUND.name());
-        } else if (payload.getProvisionResponseStatus() == TransportProtos.ProvisionResponseStatus.FAILURE) {
+            result.addProperty("status", ProvisionResponseStatus.NOT_FOUND.name());
+        } else if (payload.getStatus() == TransportProtos.ProvisionResponseStatus.FAILURE) {
             result.addProperty("errorMsg", "Failed to provision device!");
-            result.addProperty("provisionDeviceStatus", ProvisionResponseStatus.FAILURE.name());
+            result.addProperty("status", ProvisionResponseStatus.FAILURE.name());
         } else {
             if (toGateway) {
                 result.addProperty("id", requestId);
             }
-            result.addProperty("deviceId", new UUID(payload.getDeviceCredentials().getDeviceIdMSB(), payload.getDeviceCredentials().getDeviceIdLSB()).toString());
-            result.addProperty("credentialsType", payload.getDeviceCredentials().getCredentialsType().name());
-            result.addProperty("credentialsId", payload.getDeviceCredentials().getCredentialsId());
+            result.addProperty("credentialsType", payload.getCredentialsType().name());
             result.addProperty("credentialsValue",
-                    StringUtils.isEmpty(payload.getDeviceCredentials().getCredentialsValue()) ? null : payload.getDeviceCredentials().getCredentialsValue());
-            result.addProperty("provisionDeviceStatus", ProvisionResponseStatus.SUCCESS.name());
+                    StringUtils.isEmpty(payload.getCredentialsValue()) ? null : payload.getCredentialsValue());
```

Comment:
Please make a switch based on the credentials type and write the corresponding serialization here. For Basic MQTT you should write all three fields separately.